src/platform/src/main/java/org/geoserver/platform/ModuleStatusImpl.java

```diff
@@ -0,0 +1,124 @@
+/* (c) 2016 Open Source Geospatial Foundation - all rights reserved
+ * This code is licensed under the GPL 2.0 license, available at the root
+ * application directory.
+ */
+package org.geoserver.platform;
+
+/**
+ * Bean used to register module installation in applicationContext.xml.
+ * <p>
+ * Bean completly defined by applicationContext.xml - no dynamic content.
+ * <pre>
+ * &lt!-- code example needed --&gt;>
+ * </pre>
+ * @author Morgan Thompson - Boundless
+ */
+public class ModuleStatusImpl implements ModuleStatus {
+
+    private String module;
+    private String component;
+    private String name;
+    private String version;
+    private String documentation;
+    private String message;
+    
+    public ModuleStatusImpl() {
+    }
+    public ModuleStatusImpl(ModuleStatus status){
+        this.module = status.getModule();
+        this.component = status.getComponent();
+        this.name = status.getName();
+        this.version = status.getVersion();
+        this.documentation = status.getDocumentation();
+        this.message = status.getMessage();
+    }
+
+    public ModuleStatusImpl(String description) {
+        String split[] = description.split(",");
+        module = split[0];
+        name = split[1];
+    }
+    
+    /** 
+     * @return the machine readable name
+     */
+    public String getModule() {
+        return module;
+    }
+
+    /** 
+     * @param module the module name to set
+     */
+    public void setModule(String module) {
+        this.module = module;
+    }
+
+    public String getComponent() {
+        return component;
+    }
+
+    /**
+     * @param component the component to set
+     */
+    public void setComponent(String component) {
+        this.component = component;
+    }
+
+    /**
+     * @return the name
+     */
+    public String getName() {
+        return name;
+    }
+
+    /**
+     * @param name the name to set
+     */
+    public void setName(String name) {
+        this.name = name;
+    }
+
+    /**
+     * @return the version
+     */
+    public String getVersion() {
+        return version;
```

Comment:
Since this may not have been initialized it could be `null`.  The interface doesn't give any warning about that via docs or `@Nullable` annotation.  I'd prefer to see a clear requirement in the interface that these methods not return `null` with this implementation being updated to match, use the new Java 8 `Optional` object, or clearly mark them as returning `null` so that people using the value are warned that they need to do null checks.